interop/xds/server/server.go

```diff
@@ -78,8 +83,47 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for headerValue := range getRPCBehaviorMetadata(ctx) {
+		if rpcStatus := getStatusForRPCBehaviorMetadata(headerValue); rpcStatus != nil {
+			return response, status.Err(rpcStatus.Code(), rpcStatus.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) map[string]bool {
+	var mdRPCBehavior []string
+	if md, ok := metadata.FromIncomingContext(ctx); !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	} else {
+		mdRPCBehavior = md.Get(xdsinterop.RPCBehaviorMetadataKey)
+	}
+
+	rpcBehaviorMetadata := make(map[string]bool)
+	for _, val := range mdRPCBehavior {
+		splitVals := strings.Split(val, ",")
+
+		for _, behavior := range splitVals {
+			rpcBehaviorMetadata[strings.TrimSpace(behavior)] = true
```

Comment:
When you switch to a deterministic slice, I don't think trimming white space is speced anywhere/present in other languages. Please drop this trimming.

Reply:
I think I had added it because of the java implementation - https://github.com/grpc/grpc-java/blob/master/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java#L64

and they are also omitting empty strings

do the java repo owners need to be informed to remove it as well?
cc @easwars

Reply:
Commented on their implementation about whitespace: https://github.com/grpc/grpc-java/commit/78067cbace7a745fb9953aba153c13573a71024a#r126228081.